skimage/filters/thresholding.py

```diff
@@ -1177,3 +1178,69 @@ def threshold_multiotsu(image, classes=3, nbins=256):
     thresh = bin_centers[thresh_idx]
 
     return thresh
+
+
+def threshold_phansalkar(image, window_size=15, k=0.25, r=None, p=2.0, q=10.0):
+    """Applies Phansalkar local threshold to an array. Phansalkar is a
+    modification of Sauvola technique to deal with low contrast images.
+
+    This method is using the following formula::
+        T = m(x,y) * (1 + p * exp( -q * m(x,y) ) + k * ((s(x,y) / R) - 1))
+    where m(x,y) and s(x,y) are the mean and standard deviation of
+    pixel (x,y) neighborhood defined by a rectangular window with size w
+    times w centered around the pixel. k, p and q are configurable parameters.
+    R is the maximum standard deviation of a greyscale image.
+
+    Parameters
+    ----------
+    image : ndarray
+        Input image.
+    window_size : int, or iterable of int, optional
+        Window size specified as a single odd integer (3, 5, 7, …),
+        or an iterable of length ``image.ndim`` containing only odd
+        integers (e.g. ``(1, 5, 5)``).
+    k : float, optional
+        Value of the positive parameter k.
+    r : float, optional
+        Value of R, the dynamic range of standard deviation.
+        If None, set to the half of the image dtype range.
+    p : float, optional
+        Value of the parameter p.
+    q : float, optional
+        Value of the parameter q.
+
+    Returns
+    -------
+    threshold : (N, M) ndarray
+        Threshold mask. All pixels with an intensity higher than
+        this value are assumed to be foreground.
+
+    Notes
+    -----
+    This algorithm is originally designed for detection of cell nuclei in low
+    contrast images. Therefore the historgram has to be equalized beforehand
+    using :func:`skimage.exposure.equalize_adapthist`.
+
+    References
+    ----------
+    .. [1] Phansalskar N. et al. "Adaptive local thresholding for detection of
+           nuclei in diversity stained cytology images.", International
+           Conference on Communications and Signal Processing (ICCSP),
+           pp. 218-220, 2011
+           :DOI:`10.1109/ICCSP.2011.5739305`
+
+    Examples
+    --------
+    >>> from skimage import data
+    >>> from skimage.exposure import equalize_adapthist
+    >>> image = data.moon()
+    >>> image_eq = equalize_adapthist(image)
```

Comment:
Why not include this histogram equalization in the function since it seems that it is required for the method to work?

Reply:
I just followed the threshold_sauvola() fuction (and most of the other thresholding functions as well). They all return a threshold mask instead of a binary image. (Is there actually any reason for that?)
Since image_eq is necessary to create the correct binary image at the end, the function should best output a binary image directly, or what do you think?

I can see one advantage not to include the histogram equalization into the thresholding function:
If you want to find the best parameters window_size, k, r, p, q for a special image set, it is much faster if you have to calculate equalize_adapthist() only once, instead every time you change a parameter.
For example:

```
def threshold_phansalkar_parameter_variation( image ):
    #equalize only once
    image_eq = equalize_adapthist(image)
    
    # list of parameters
    k_list = [0.2,0.25,0.3,0.35]
    p_list = [1.5,2.0,2.5,3.0]

    fig, ax = plt.subplots(len(p_list),len(k_list), figsize=(15, 6) )
    fig.subplots_adjust(hspace = .5)
    ax = ax.ravel()

    for j in range(len(p_list)):
        for i in range(len(k_list)):
            # calculate binary mask of thresh_phansalkar() without doing equalize_adapthist() every time
            thresh_phansalkar = threshold_phansalkar(image_eq, window_size=15, k=k_list[i], p=p_list[j], q=10.0)
            binary_phansalkar = image_eq > thresh_phansalkar

            ax[i+j*len(k_list)].imshow(binary_phansalkar, cmap=plt.cm.gray)
            ax[i+j*len(k_list)].set_title( "k = " + str( k_list[i] )+ ", p = " + str( (p_list[j]) ) )
            ax[i+j*len(k_list)].set_axis_off()

    plt.show()
```

If you think that this is too rare a usecase, then your suggestion would definitely be more user-friendly.

Reply:
According to @jni the function should return a threshold map. Therefore, I thing it is best to leave the function. I documented the requirements of the function in the funtion itself and the example.